compiler/rustc_resolve/src/build_reduced_graph.rs

```diff
@@ -1534,3 +1539,16 @@ impl<'a, 'b, 'tcx> Visitor<'b> for BuildReducedGraphVisitor<'a, 'b, 'tcx> {
         }
     }
 }
+
+fn check_allow_unused_imports_attr(attrs: &AttrVec) -> bool {
```

Comment:
This is really unreliable, the `allow` attribute may be e.g. on a parent module instead of the `use` item itself, and I wouldn't want to include the lint level tree logic into resolve.

In general, the `unused_qualifications` lint is not very reliable and rarely used, that's why it's allow-by-default.
So, I think we can prioritize the unused import lints and just do some minimal repair to the `unused_qualifications` lint so it doesn't produce breaking suggestions.

So, if we don't consider `allow` we would still suppress `unused_qualifications` even if it refers to an allowed unused imports, and that seems fine to me.